pkg/bot/discord.go

```diff
@@ -75,19 +83,33 @@ func NewDiscord(log logrus.FieldLogger, commGroupName string, cfg config.Discord
 	}
 
 	return &Discord{
-		log:             log,
-		reporter:        reporter,
-		executorFactory: executorFactory,
-		api:             api,
-		botID:           cfg.BotID,
-		commGroupName:   commGroupName,
-		channels:        channelsCfg,
-		botMentionRegex: botMentionRegex,
-		renderer:        NewDiscordRenderer(),
+		log:                   log,
+		reporter:              reporter,
+		executorFactory:       executorFactory,
+		api:                   api,
+		botID:                 cfg.BotID,
+		commGroupName:         commGroupName,
+		channels:              channelsCfg,
+		botMentionRegex:       botMentionRegex,
+		renderer:              NewDiscordRenderer(),
+		messages:              make(chan discordMessage, discordMessageChannelSize),
+		discordMessageWorkers: pool.New().WithMaxGoroutines(discordMessageWorkersCount),
 	}, nil
 }
 
-// Start starts the Discord websocket connection and listens for messages.
+func (b *Discord) startMessageProcessor(ctx context.Context) {
+	b.log.Info("Starting discord message processor...")
+	defer b.log.Info("Stopped discord message processor...")
+
+	for msg := range b.messages {
+		b.discordMessageWorkers.Go(func() {
+			err := b.handleMessage(ctx, msg)
+			if err != nil {
+				b.log.Errorf("while handling message: %s", err.Error())
```

Comment:
```suggestion
				b.log.WithError(err).Error("Failed to handle Discord message")
```
that's actually a log message, not error wrapping.